discovery/gossiper.go

```diff
@@ -2370,91 +2367,113 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(
 	}
 }
 
-// sendAnnSigReliably will try to send the provided local AnnounceSignatures
-// to the remote peer, waiting for it to come online if necessary. This
-// method returns after adding the message to persistent storage, such
-// that the caller knows that the message will be delivered at one point.
-func (d *AuthenticatedGossiper) sendAnnSigReliably(
-	msg *lnwire.AnnounceSignatures, remotePeer *btcec.PublicKey) error {
-
-	// We first add this message to the database, such that in case
-	// we do not succeed in sending it to the peer, we'll fetch it
-	// from the DB next time we start, and retry. We use the peer ID
-	// + shortChannelID as key, as there possibly is more than one
-	// channel opening in progress to the same peer.
-	var key [41]byte
-	copy(key[:33], remotePeer.SerializeCompressed())
-	binary.BigEndian.PutUint64(key[33:], msg.ShortChannelID.ToUint64())
+// sendMsgReliably attempts to send the message to the peer.
+func (d *AuthenticatedGossiper) sendMsgReliably(msg lnwire.Message,
+	peerKey *btcec.PublicKey) error {
 
+	// First, we'll add the message to the database to ensure we can resend
+	// it upon restarts if the peer has yet to come back online. Each
+	// message is keyed by a tuple of the peer's public key and the message
```

Comment:
As mentioned in the removed comment, the short channel ID must be part of the key, to handle the case with more than one channel to the same peer.